### ISSUE TITLE
LLOGO 340 display turtle problems

### ISSUE THREAD

**Issue by larsbrinkhoff** [PDP-10/its#1433]
LLOGO doesn't work with the 340 display turtle:

```
*llogo^H
!
LISP 2154
LLOGO 1700
DO YOU WANT TO USE THE DISPLAY TURTLE?
<y
DO YOU WANT TO USE THE GT40 RATHER THAN THE 340?
<n
LLOGO LISTENING
?cs
;COULDN'T EVALUATE WIPE NIL
;BECAUSE DISCRIBE IS AN UNDEFINED PROCEDURE
```

I'm guessing this is because LISP; SLAVE FASL isn't loaded.  DISCRIBE is defined there.

**Comment by larsbrinkhoff** [comment]
But even if it's loaded, the 340 slave has a problem:

```
l^H
!
LISP 2154
Alloc? n
*(load "lisp; slave")
;Loading SLAVE 8
127303
(disini)
PDP6 NOT AVAILABLE TRYING PDP10 SLAVE
BADPI;(ILOPR;) 212>>0

**Comment by larsbrinkhoff** [comment]
@eswenson1, do you know how to debug this?  I tried setting breakpoints in SLAVE FASL, but DDT doesn't know about the symbols.  ($^K doesn't help.)

**Comment by eswenson1** [comment]
I don't. But you could assemble SLAVE and produce. a listing, and using two SUPDUP sessions, side-by-side, bring up the listing in emacs in one and debug without symbols in the other. I've done that before and once you have "found" where you are, it is pretty easy to debug.  Are you sure the ILOPR is happening as a result of code in SLAVE FASL, or might the ILOPR be happening in the interpreter/runtime?

**Comment by larsbrinkhoff** [comment]
Thanks, I'll try that.

I'm not sure about anything yet.  "PDP6 NOT AVAILABLE TRYING PDP10 SLAVE" is a message that comes from SLAVE FASL, and the ILOPR comes right after that.

**Comment by larsbrinkhoff** [comment]
I did this to find the start of the string " TRYING PDP10 SLAVE": `6462715156$W`  Then this to find the instruction to print the string: `7000xxxxxx$W` where the x'es are that address.

After printing the string, OPNTEN is called.  It opens an inferior job and loads SYS; ATSIGN 10SLAV.  Shortly after this, I get: `BADPI;(ATTY;)`  I'm not sure what to make of this, or where it's coming from.

**Comment by larsbrinkhoff** [comment]
The inferior DSLAVE job is .HANGing waiting for FUNCTION to be set.  Looks ok.

**Comment by eswenson1** [comment]
I would guess that the DSLAVE job issued an interrupt and the superior process had not set up a user interrupt handler (in location 42?) to handle that interrupt.  Thus, ITS interrupted the superior (LISP) and gave it a BADPI interrupt.

**Comment by larsbrinkhoff** [comment]
A reasonable guess, but I don't see that the slave is doing anything untoward.  It's just initializing and then waiting for a command with .HANG.

```
*$l sys; atsign 10slav
*$0g
GO>>SETZM FUNCTI   FUNCTI/   0      ^N
GO+1>>SETZM DISLIS   DISLIS/   0      ^N
REGO>>MOVE P,CONST   P/   0   CONST/   -310,,CONST+41      ^N
REGO+1>>SETZM ERRLOC   ERRLOC/   0      ^N
REGO+2>>SETOM DENABL   DENABL/   -1      ^N
REGO+3>>SETZM SCPSTE   SCPSTE/   0      ^N
REGO+4>>PUSHJ P,DSTART      $^N
REGO+5>>742200,,40   40/   0      ^N
REGO+6>>SETZM DBRITE   DBRITE/   0      ^N
REGO+7>>SETZM DSCALE   DSCALE/   0      ^N
REGO+10>>SETZM PENPOS   PENPOS/   0      ^N
REGO+11>>SETZM ASTATE   ASTATE/   0      ^N
REGO+12>>SETZM ARYNUM   ARYNUM/   0      ^N
REGO+13>>SETOM SUPFLG   SUPFLG/   -1      ^N
REGO+14>>MOVE W,CONST+1   W/   0   CONST+1/   JRST GETMOD      ^N
REGO+15>>MOVEM W,PSWIT   W/   JRST GETMOD   PSWIT/   JRST GETMOD      ^N
REGO+16>>MOVE W,CONST+2   W/   JRST GETMOD   CONST+2/   MSKIP      ^N
REGO+17>>MOVEM W,MODTBL+1   W/   MSKIP   MODTBL+1/   MSKIP      ^N
MAIN0>>SKIPN FUNCTI   FUNCTI/   0      ^N
MAIN0+1>>.HANG   0/   0

**Comment by larsbrinkhoff** [comment]
I checked the contents of 212 before calling `disini`.  It was not 0.  I tried to check writes to this location with `212$2I`, but I didn't see anything.  I think it may be the CORBLK in OPNTEN (.CALL SIXMAP) that could be wreaking havoc with the Lisp memory image.

**Comment by larsbrinkhoff** [comment]
.CALL SIXMAP uses the location at SIXPG to specify the page number in the Lisp core image.  When TENOPN is called, that location is 0, which is no good.  It's supposed to be set in OP6D2 from a call to GETCOR.  This is only called from OPNSIX, not OPNTEN.

**Comment by larsbrinkhoff** [comment]
Updated TENOPN to call GETCOR, as in #1461.  Now LLOGO works with the 340 turtle, if SLAVE FASL is loaded manually.

**Comment by larsbrinkhoff** [comment]
Fixed by #1461.
